salt/_modules/better_kubernetes.py

```diff
@@ -0,0 +1,148 @@
+"""Execution methods for management of Kubernetes objects.
+
+This module relies on the `kubernetes` custom Salt utils module for parsing
+K8s object manifests, and providing direct bindings to the Python `kubernetes`
+client.
+
+TODO:
+- handle custom objects
+- add some logging for debug purposes
+- add some documentation for each method
+- support giving a `path` to a file instead of directly the `manifest` data
+- improve management of `kubeconfig` and `context` parameters, relying on
+  master configuration and sane defaults - look into `__opts__`
+"""
+from salt.exceptions import CommandExecutionError
+
+try:
+    import kubernetes.client as k8s_client
+except ImportError:
+    HAS_LIBS = False
+else:
+    HAS_LIBS = True
+
+__virtualname__ = 'metalk8s_kubernetes'
+
+
+def __virtual__():
+    if not HAS_LIBS:
+        return False, 'Missing Python `kubernetes.client` dependency'
+
+    return __virtualname__
+
+
+def _extract_obj_and_details(manifest):
+    try:
+        details = __utils__['kubernetes.get_object_details'](manifest)
+        obj = __utils__['kubernetes.convert_manifest_into_object'](manifest)
+    except ValueError as exc:
+        raise CommandExecutionError('Invalid manifest: {!s}'.format(exc))
```

Comment:
So, what I understand of this `CommandExecutionError` is that it's the standard one to raise from **execution modules**, so that invoking `salt 'target' my_module.my_command` will output a proper message. This is just a pattern I observed, so not entirely sure about it. Neither is it documented in [Writing Execution Modules](https://docs.saltstack.com/en/latest/ref/modules/), nor anything related to raising exceptions from them.

Any more info on this @TeddyAndrieux @sayf-eddine-scality @NicolasT ?

Reply:
All I can say is: I've seen `CommandExecutionError` being used all over the place in Salt modules as well, so this seems legit.

Reply:
Ok, let's keep it then :slightly_smiling_face: